### ISSUE TITLE
Changing a classes property in code

### ISSUE THREAD

**Issue by travtrigs** [felixhageloh/uebersicht#118]
TL;DR at bottom

I'm making a widget that displays the current location of the planets in our solar system. Currently I have 9 image files of equal dimensions (1000x1000). One for the sun (centered), and one for every planet (located relative to the center). When I display them all stacked on top of each other, it creates a complete image of the solar system (sun is the yellow dot at the center)
<img width="825" alt="screen shot 2015-08-09 at 2 41 17 am" src="https://cloud.githubusercontent.com/assets/13620298/9153837/2c93d410-3e40-11e5-96bd-24ab31d6e6e1.png">
I use the command "date '+%j' %Y" to get the current day of the year, and the year. Then I split the output    and store the day and the year. Since I made the images of the planets using their locations as of yesterday (August 8, 2015, day 220), I subtract 220 from day and 2015 from year. Then I multiply year by 365 and add the result to day. This gives me the number of days since I made the images. Today it would be 2, a year from today it would be 367. Then I define an array (named "constants") with values that correspond to how many degrees each planet should move for every day that passes on earth. Finally, I create another array (named "rotations") containing the values of the first array multiplied by the current day. This array represents how many degrees each planet should rotate.
I want to use those values to rotate each image. So if the first and second elements are 12 and 50, the image of Mercury and Venus should be rotated by 12 and 50 degrees, respectively. I know that I can rotate an image using "transform: rotate(#deg)", but I can't figure out how to pass the values of "rotations" to the transform property. How can I accomplish that? What I tried to do originally was "transform: rotate(@rotations[index])", but that didn't work. Here's my code: https://github.com/travtrigs/solar-system.widget/blob/master/index.coffee

TL;DR How can I have a variable in a property, specifically "transform: rotate(#{x}deg)"

**Comment by felixhageloh** [comment]
Cool widget idea!
The syntax you are using is fine, it is just that 
a) the styles are compiled before the widget is instantiated. That means you can use closure variables but not instance variables when interpolating styles.
b) styles are evaluated only once and remain static after that

So to solve your problem, you'll have to update the css using javascript like so:
```coffeescript
$('.Earth').css transform: "rotate(#{@rotations.earth}deg)"
```

If you keep your rotations in a hash and use dom ids that match the hash keys, you can do this quite nicely in coffeescript (pseudocode):

```coffeescript 
@rotations = 
  mercury: ..
  venus: ...
  earth: ..

for planet, rotation of @rotations
  $("##{planet}").css transform: "rotate(#{rotation}deg)" 
```

**Comment by travtrigs** [comment]
I tried using      $(‘.Mercury’).css transform: “rotate(90deg)”      just to see if it would work, but even that doesn’t work. I updated my code (https://github.com/travtrigs/solar-system.widget/blob/master/index.coffee) to show what I tried.

**Comment by felixhageloh** [comment]
I just noticed: Your update method returns a string, while it should just be code. So if you get rid of the quotes around the code of your update method, like

```coffeescript
update: (output, domEl) ->
  date = output.split(/\s+/)
  day = (date[0]-220) + (date[1]-2015)*365
  #doesn't work
  $('.Mercury').css transform: "rotate(90deg)"
  @rotations =
    mercury: 4.09090909090909
    venus: 1.6
    earth: 0.9863013698630137
    mars: 0.52401746724891
    saturn: 0.08310249307479
    jupiter: 0.0334603587694
    uranus: 0.01176893654582
    neptune: 0.00602006688963
  #gives error "Can't find variable: planet"
  for planet, rotation of @rotations
    $("##{planet}").css transform: "rotate(#{rotation*day}deg)"

```

it should work

**Comment by travtrigs** [comment]
Finally!! Thanks so much! When I use “for planet, rotation of @rotations” it doesn’t work though. I could definitely get this working without using it and just use        $('.Mercury').css transform: "rotate(90deg)” because that DOES work, but if I could get the for loop working that would be much better.

**Comment by felixhageloh** [comment]
you'll have to change your HTML slightly:

```coffeescript
render: ->"""
  <img id="sun" src="solar-system.widget/Sun.png"/>
  <img id="mercury" src="solar-system.widget/Mercury.png"/>
  <img id="venus" src="solar-system.widget/Venus.png"/>
  ...
"""
```

The for loop assumes dom ids that match the keys in your hash

**Comment by travtrigs** [comment]
Okay, I tried that, however, now only the “sun” image is displayed, none of the planets appear.

**Comment by travtrigs** [comment]
It seems that using id instead of class then makes it ignore the CSS, and renders only the first image.

**Comment by travtrigs** [comment]
Got it! Just had to replace all the “.”s in the CSS with “#”s. Thanks so much for the help! Have been searching for a solution for days, can finally get this finished.
